src/client/csfle/client_encryption.rs

```diff
@@ -466,11 +575,47 @@ impl<'a> EncryptAction<'a> {
     }
 
     /// Set the query type.
-    #[allow(clippy::redundant_clone)]
     pub fn query_type(mut self, qtype: impl Into<String>) -> Self {
         self.opts.query_type = Some(qtype.into());
         self
     }
+
+    /// NOTE: This method is experimental and not intended for public use.
+    ///
+    /// Set the range options. It is an error to call this method when the algorithm is not
+    /// [`Algorithm::RangePreview`].
+    pub fn range_options(mut self, range_options: impl Into<Option<RangeOptions>>) -> Self {
```

Comment:
Because this will be enforced by libmongocrypt I'd prefer to leave our builder methods infallible and keep our logic simple. (This was the general principle I followed with the other possible invalid state, e.g. specifying precision without min/max.) That said I think my comment is confusing given the fact that this method can't actually return an error, so I can update that to be more clear if this sounds good to you.

Reply:
Yup, makes sense.